linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml

```diff
@@ -23,69 +27,276 @@
 
-- name: "start relevant networks"
-  virt_net:
-    state: active
-    name: "{{ net_item['name'] }}"
-  with_items:
-    - "{{ res_def['networks'] }}"
+  # note: though default_ssh_key_path is fetched from linchpin.conf , default value is needed to make playbooks work independently regardless of linchpin.conf
+- name: "local: does ssh key already exists ?"
+  stat:
+    path: "{{ default_ssh_key_path | default('~/.ssh/') }}{{ res_def['ssh_key'] | default(res_grp['resource_group_name'])}}"
+  register: ssh_key_stat_local
+
+- name: "remote: does ssh key already exists ?"
+  stat:
+    path: "{{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] | default(res_grp['resource_group_name'])}}"
+  register: ssh_key_stat_remote
+  remote_user: "{{ res_def['remote_user'] | default('root') }}"
+  delegate_to: "{{ uri_hostname }}"
+  when: uri_hostname != 'localhost'
+
+- name: "local: Generate ssh keys when they donot exist"
+  shell: "ssh-keygen -t rsa -f {{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] | default(res_grp['resource_group_name']) }} -N '';cat {{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] | default(res_grp['resource_group_name']) }} >> ~/.ssh/authorized_keys"
+  args:
+    creates: "{{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] }}"
+  when: uri_hostname =='localhost' and ssh_key_stat_local.stat.exists == false
+
+- name: "remote: Generate ssh keys when they donot exist"
+  shell: "ssh-keygen -t rsa -f {{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] | default(res_grp['resource_group_name']) }} -N '';cat {{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] }} >> ~/.ssh/authorized_keys"
+  args:
+    creates: "{{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] }}"
+  remote_user: "{{ res_def['remote_user'] | default('root') }}"
+  delegate_to: "{{ uri_hostname }}"
+  when: uri_hostname != 'localhost' and ssh_key_stat_remote.stat.exists == false
+
+- name: "local: cat out contents of public key"
+  shell: "cat {{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] | default(res_grp['resource_group_name']) }}.pub"
+  register: pubkey_local
+  when: uri_hostname == 'localhost'
+
+- name: "remote: cat out contents of public key"
+  shell: "cat {{ default_ssh_key_path | default('~/.ssh/') }}/{{ res_def['ssh_key'] | default(res_grp['resource_group_name']) }}.pub"
+  register: pubkey_remote
+  remote_user: "{{ res_def['remote_user'] | default('root') }}"
+  delegate_to: "{{ uri_hostname }}"
+  when: uri_hostname != 'localhost'
+
+- name: "Add additional storage"
+  command: "qemu-img resize {{ definition[1] }}/{{ definition[0] }}_{{ definition[4] }}.{{ definition[2] }} +{{ definition[3] }}G"
+  with_nested:
+    - ["{{ libvirt_resource_name }}"]
+    - ["{{ libvirt_image_path }}"]
+    - ["{{ img_src_ext }}"]
+    - ["{{ res_def['additional_storage'] | default('1') }}"]
```

Comment:
This feels like good option. But it doesn't appear to be in the schema? Maybe update that so it checks for it appropriately?